### PR TITLE
fix(#2074): VaSwitch made input clickable and fire change event

### PR DESCRIPTION
**Pull request by m0ksem**
closes https://github.com/epicmaxco/vuestic-ui/issues/2074

### CODE REVIEW

File: packages/ui/src/components/va-switch/VaSwitch.stories.ts [hunk 1]
```diff
@@ -1,12 +1,51 @@
...
+  methods: {
+    onChange (e) {
+      if (e.target) {
+        this.changeCount++
+      }
+    },
+  },
+
+  template: `
+    [changes count]: {{ changeCount }}
```
Comment (asvae) [inline packages/ui/src/components/va-switch/VaSwitch.stories.ts:32]: small thing, but the convention is something like in Backtop > ClickEvent.

```html
<p>count: <span data-testid="count">{{ count }}</span></p>
```

```typescript
(canvas.getByTestId('clicked').innerText).toBe('2')
```

File: packages/ui/src/components/va-switch/VaSwitch.stories.ts [hunk 2]
```diff
@@ -1,12 +1,51 @@
...
+
+  await userEvent.click(input)
+
+  expect(e.canvasElement.textContent).toContain('[changes count]: 1')
+
+  await userEvent.keyboard(' ')
+
+  expect(e.canvasElement.textContent).toContain('[changes count]: 2')
+
+  await userEvent.type(input, '{enter}')
```
Comment (asvae) [inline packages/ui/src/components/va-switch/VaSwitch.stories.ts:48]: We don't need to test for enter.

File: packages/ui/src/components/va-switch/VaSwitch.stories.ts [hunk 3]
```diff
@@ -1,12 +1,51 @@
...
 })
+
+export const ChangeEvent: StoryFn = () => ({
+  components: { VaSwitch },
+
+  data: () => ({ value: false, changeCount: 0 }),
+
+  methods: {
+    onChange (e) {
+      if (e.target) {
```
Comment (asvae) [inline packages/ui/src/components/va-switch/VaSwitch.stories.ts:25]: Why do we need this check?

File: packages/ui/src/components/va-switch/VaSwitch.stories.ts [hunk 4]
```diff
@@ -1,12 +1,51 @@
...
+})
+
+ChangeEvent.play = async (e) => {
+  const input = e.canvasElement.querySelector('input')!
+
+  await userEvent.click(input)
+
+  expect(e.canvasElement.textContent).toContain('[changes count]: 1')
+
+  await userEvent.keyboard(' ')
```
Comment (asvae) [inline packages/ui/src/components/va-switch/VaSwitch.stories.ts:44]: Pretty sure this is handled by click event, no?

**Review by asvae — APPROVED** [review]
Functionality side is fine. I hope to see story improved a bit.

File: packages/ui/src/components/va-switch/VaSwitch.stories.ts [hunk 5]
```diff
@@ -1,12 +1,51 @@
...
   template: '<VaSwitch/>',
 })
+
+export const ChangeEvent: StoryFn = () => ({
+  components: { VaSwitch },
+
+  data: () => ({ value: false, changeCount: 0 }),
+
+  methods: {
+    onChange (e) {
```
Comment (asvae) [inline packages/ui/src/components/va-switch/VaSwitch.stories.ts:24]: Would be nice to check event value as well, not only count.

File: packages/ui/src/components/va-switch/VaSwitch.stories.ts [hunk 6]
```diff
@@ -1,12 +1,51 @@
...
-  components: { VaSwitch },
+export const Default = () => defineComponent({
+  components: { VaSwitch: VaSwitchDemo },
   template: '<VaSwitch/>',
 })
+
+export const ChangeEvent: StoryFn = () => ({
+  components: { VaSwitch },
+
+  data: () => ({ value: false, changeCount: 0 }),
```
Comment (asvae) [inline packages/ui/src/components/va-switch/VaSwitch.stories.ts:21]: We don't strictly need value and v-model here.

File: packages/ui/src/components/va-switch/VaSwitch.stories.ts (hunk 1 above)
Comment (m0ksem) [reply packages/ui/src/components/va-switch/VaSwitch.stories.ts:32]: querySelector seems much more clean solution for me, than within and getByTestId. I'd prefer querySelector.

**Pull request merged**